rocky/rocky/scheduler.py

```diff
@@ -189,21 +189,35 @@ class TaskNotFoundError(SchedulerError):
     message = _("Task could not be found.")
 
 
-class SchedulerClient:
+class SchedulerSession(requests.Session):
     def __init__(self, base_uri: str):
-        self.session = requests.Session()
+        super().__init__()
+        self.session = self
         self._base_uri = base_uri
-        self.is_alive()
 
+        try:
+            self.session.get(f"{self._base_uri}/health")
```

Comment:
These kind of calls do *not* belong in a constructor/ initializer. We've talked about doing these kind of calls before

Reply:
We need to somehow talk to the Scheduler, before the lazyloading happens, it will not happen if its not checked in init. I think getting the /health endpoint or just / is enough to check if its online.

Reply:
What do you mean by we *need to*? Lazy loading must happen when it needs to happen, hence *lazy loading*. Health checking is for troubleshooting or startup requirements, **not a requirement to check for every request**

Reply:
@Rieven means we need to figure out if the scheduler is available or not before we do a lazy loading call. A lazy loading call would end up being executed in the template context where we have a harder time handling any errors.
Since doing a health check before hand is an extra request, we should not do so, it just makes everything more expensive, and might still end up erroring out when we do an actual data request.

I think we should not make this (or any other restfull api request) lazy loading. If these requests are needed, they should be done in the model (or view), and we should handle any errors and exceptions in that scope too. Our targeted restfull api's dont offer anything that resembles lazy-loading anyway, so there's no gain in having these calls be lazy or deferred to when the template is rendering.

Reply:
Ok, then I need to refactor the lazyloader which means that I need to also refactor pagination for task items.